src/casp/services/api/services/cell_operations_service.py

```diff
@@ -170,67 +170,71 @@ def __split_embeddings_into_chunks(embeddings: np.array, chunk_size: int) -> t.L
         Annotate a single anndata file with Cellarium CAS. Input file should be validated and sanitized according to the
         model schema. Increment user cells processed counter after successful annotation.
 
         :param user: User object used to increment user cells processed counter.
         :param file: Byte object of :class:`anndata.AnnData` file to annotate.
         :param model_name: Model name to use for annotation. See `/list-models` endpoint for available models.
-        :param include_dev_metadata: Boolean flag indicating whether to include dev metadata in the response.
+        :param consensus_strategy: Consensus strategy to use for annotation from `ConsensusStrategyType` enum.
+        :param normalize: Boolean flag indicating whether to normalize the embeddings before annotation. Used only in
+            `ontology_aware` method.
+        :param include_dev_metadata: Boolean flag indicating whether to include dev metadata in the response. Used only
+            in `cell_type_count` method.
 
         :return: JSON response with annotations.
         """
+        if normalize is None and consensus_strategy == consensus_engine.ConsensusStrategyType.ONTOLOGY_AWARE:
+            raise exceptions.InvalidInputError("Normalization flag is required for Ontology Aware method.")
+        if (
+            include_dev_metadata is None
+            and consensus_strategy == consensus_engine.ConsensusStrategyType.CELL_TYPE_STATISTICS
+        ):
+            raise exceptions.InvalidInputError("Include Dev Metadata flag is required for Cell Type Count method.")
+
         self.authorize_model_for_user(user=user, model_name=model_name)
         query_ids, embeddings = await CellOperationsService.get_embeddings(file_to_embed=file, model_name=model_name)
 
         if embeddings.size == 0:
             # No further processing needed if there are no embeddings
             return []
         knn_response = await self.get_knn_matches(embeddings=embeddings, model_name=model_name)
-        annotation_response = self.get_cell_type_distribution(
-            query_ids=query_ids,
-            knn_response=knn_response,
-            model_name=model_name,
-            include_dev_metadata=include_dev_metadata,
-        )
+        cas_model = self.cellarium_general_dm.get_model_by_name(model_name=model_name)
+
+        match consensus_strategy:
+            case consensus_engine.ConsensusStrategyType.ONTOLOGY_AWARE:
+                strategy = consensus_engine.OntologyAwareConsensusStrategy(
+                    cell_operations_dm=self.cell_operations_dm,
+                    normalize=normalize,
+                )
+            case consensus_engine.ConsensusStrategyType.CELL_TYPE_STATISTICS:
+                strategy = consensus_engine.CellTypeStatisticsConsensusStrategy(
+                    cell_operations_dm=self.cell_operations_dm,
+                    cas_model=cas_model,
+                    include_dev_metadata=include_dev_metadata,
```

Comment:
this could be a kwarg for CELL_TYPE_STATISTICS strategy w/ default value set to False.